core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -706,7 +706,7 @@ class Partition(val topicPartition: TopicPartition,
           s"and partition state $partitionState since it is already a follower with leader epoch $leaderEpoch.")
       }
 
-      leaderReplicaIdOpt = Some(partitionState.leader)
+      leaderReplicaIdOpt = Option(partitionState.leader)
```

Comment:
Is it worth adding the current leader to the "Follower $topicPartition starts at ..."  log message above?

Reply:
Good idea.